core/sql/generator/GenPreCode.cpp

```diff
@@ -11274,6 +11274,485 @@ short HbaseAccess::extractHbaseFilterPreds(Generator * generator,
+	      newCV = newCV->bindNode(generator->getBindWA());
+	      newCV = newCV->preCodeGen(generator);
+
+	      foundBinary = TRUE;
+	      flipOp = TRUE;
+	      colVID = newCV->getValueId();
+	      valueVID = ie->child(0)->getValueId();
+	    }
+	}
+    }//end binary operators
+  else if (ie && ((ie->getOperatorType() == ITM_IS_NULL)||(ie->getOperatorType() == ITM_IS_NOT_NULL))){//check for unary operators
+	  ItemExpr * child0 = ie->child(0)->castToItemExpr();
+	  if ((ie->child(0)->getOperatorType() == ITM_BASECOLUMN) ||
+		  (ie->child(0)->getOperatorType() == ITM_INDEXCOLUMN)||
+		  (ie->child(0)->getOperatorType() == ITM_REFERENCE)){
+		  foundUnary = TRUE;
+		  colVID = ie->child(0)->getValueId();
+		  valueVID = NULL_VALUE_ID;
+	  }
+
+  }//end unary operators
+
+  //check if found columns belong to table being scanned (so is not an input to the scan node)
+  if (foundBinary || foundUnary){
+	ValueId dummyValueId;
+	if (getGroupAttr()->getCharacteristicInputs().referencesTheGivenValue(colVID,dummyValueId)){
+		foundBinary=FALSE;
+		foundUnary=FALSE;
+	}
+  }
+  //check if not an added column with default non null
+  if ((foundBinary || foundUnary)&& (NOT hbaseLookupPred)){
+	  NAColumn * nac;
+		switch (colVID.getItemExpr()->getOperatorType()){
+		case ITM_BASECOLUMN:
+			nac = ((BaseColumn*)colVID.getItemExpr())->getNAColumn();
+			break;
+		case ITM_INDEXCOLUMN:
+			nac = ((IndexColumn*)colVID.getItemExpr())->getNAColumn();
+			break;
+		default:
+			break;
+		}
+		if (nac && nac->isAddedColumn() && nac->getDefaultValue()){
+			foundBinary=FALSE;
+			foundUnary=FALSE;
+		}
+  }
+
+  if (foundBinary)
+    {
+      const NAType &colType = colVID.getType();
+      const NAType &valueType = valueVID.getType();
+
+      NABoolean generateNarrow = FALSE;
+      if (NOT hbaseLookupPred)
+	{
+	  generateNarrow = valueType.errorsCanOccur(colType);
+	  if ((generateNarrow)  || // value not a superset of column
+	      (NOT columnEnabledForSerialization(colVID.getItemExpr())))
+		  foundBinary = FALSE;
+	}
+
+      if (foundBinary)
+	{
+	  if (colType.getTypeQualifier() == NA_CHARACTER_TYPE)
+	    {
+	      const CharType &charColType = (CharType&)colType;
+	      const CharType &charValType = (CharType&)valueType;
+
+	      if ((charColType.isCaseinsensitive() || charValType.isCaseinsensitive()) ||
+		  (charColType.isUpshifted() || charValType.isUpshifted()))
+	     foundBinary = FALSE;
+	    }
+	  else if (colType.getTypeQualifier() == NA_NUMERIC_TYPE)
+	    {
+	      const NumericType &numType = (NumericType&)colType;
+	      const NumericType &valType = (NumericType&)valueType;
+	      if (numType.isBigNum() || valType.isBigNum())
+	     foundBinary = FALSE;
+	    }
+	}
+
+      if (foundBinary)
+	{
+	  if ((ie) && (((BiRelat*)ie)->addedForLikePred()) &&
+	      (valueVID.getItemExpr()->getOperatorType() == ITM_CONSTANT))
+	    {
+	      // remove trailing '\0' characters since this is being pushed down to hbase.
```

Comment:
I'm guessing that HBase doesn't use null-termination? It might make sense to take one trailing '\0', but is it possible we are sending down a binary string with nulls in it? I'm wondering if this code will handle that correctly by removing all trailing '\0's?  (But maybe HBase has null-padding semantics? If it does then this should not matter.)